frontend/packages/knative-plugin/src/components/traffic-splitting/TrafficSplittingController.tsx

```diff
@@ -1,27 +1,9 @@
 import * as React from 'react';
 import { K8sResourceKind } from '@console/internal/module/k8s';
-import { Firehose, FirehoseResult } from '@console/internal/components/utils';
+import { Firehose } from '@console/internal/components/utils';
 import { createModalLauncher, ModalComponentProps } from '@console/internal/components/factory';
-import {
-  transformTrafficSplitingData,
-  knativeServingResourcesTrafficSplitting,
-} from '../../utils/traffic-splitting-utils';
-import TrafficSplitting from './TrafficSplitting';
-
-type ControllerProps = {
-  loaded?: boolean;
-  obj: K8sResourceKind;
-  resources?: {
-    configurations: FirehoseResult;
-    revisions: FirehoseResult;
-  };
-};
-
-const Controller: React.FC<ControllerProps> = (props) => {
```

Comment:
What was the need to move the `Controller` component into a separate file? You are not using the component anywhere else. And now you have two components `Controller` and `TrafficSplitingController` which seems ambiguous to me.

Reply:
Agreed. In this case the only reason we have this other component is because of the poor implementation of `Firehose` which injects props into the child component. We will have (already do?) hooks to perform firehose like behavior and this component will just go away then.

Reply:
The intention was to follow "one file on component", but I can see now that this component just exists for implementing `Firehose` and has no other use. Have reverted to the initial implementation.